### ISSUE TITLE
Ideas & Observations

### ISSUE THREAD

**Issue by yoshimo** [bleak-lodge/plugin.video.whitelodge#1]
This is  a very nice idea.

For the future i think it would be great to:
1. sync watched episodes and movies with the original service and your personal justwatch account. This makes it robust and makes sure everything you watched will be part of the reccommendation system.
2. Get the typical design with horizontal movies and episodes and the categories vertically with preview pictures that we are used to from the offical apps and websites. Last seen things pinned on top for continued watching.
3. If you are looking for things that happen to be on multiple services, use the one with the best quality automatically.
This includes, different audio tracks and higher resolution as well as subtitles and if it has to be rented, bought or is part of the flatrate
4. Sometimes i get search results, but when i click it, it doesn't find any source on either netflix, disney, hbo ,prime . This is kind of confusing.
Are these things that i would have to pay additionally for which amazon prime plugin would mark as "orange" ?

**Comment by bleak-lodge** [comment]
Thanks for your input.
I edited your post a bit, numbering your multiple queries so that I can more easily address them.

1. This add-on doesn't interact directly with any of the supported streaming services. It does interact with JustWatch, specifically through an unofficial API I've found for it, and through it's collaboration with TMDb. Unfortunately none of these paths offer any kind of personalised data (as user's watch states) whatsoever. The add-on uses Trakt for a more broad way to set/get watched states.
2. That's really a skin matter. Once you've found your preferred skin and view type of it, you should go to Whitelodge/Tools/Viewtypes and save that view for each content type, so that it sticks for every menu of the add-on. For the "categories vertically with preview pictures" it's not possible within the add-on's menus, I think you should use a skin with widgets support.
3. I'll look into implementing a preferred service system (if more than 1 enabled) but I don't think there's a way for me to know which has the best stream - let alone audio tracks, subtitles etc.
4. You mean through the add-on's Search menu? Yea that's not limiting results based on enabled services, it's a broad search on TMDb's database. I'm working on a solution that will filter the search results based on user's enabled services though. Also, there are cases where despite being on a specific services menu, some items might not have streams available as well. The JustWatch database unfortunately isn't perfect - it has shows missing from services, or others that are not still available on a service.

**Comment by yoshimo** [comment]
Wouldn't we be able to improve Point 4 by working together with the authors of the downstream plugins?
Move from a TMDB search to an aggregated search on the individual plugins may increase the accuracy in case of JW failing.

**Comment by psychofaktory** [comment]
Another idea for a future function:
When scheduling a TV recording via a PVR addon, the whitelodge plugin could check whether the programme to be recorded is already available via one of the subscribed streaming services.

**Comment by bleak-lodge** [comment]
> Wouldn't we be able to improve Point 4 by working together with the authors of the downstream plugins?
> Move from a TMDB search to an aggregated search on the individual plugins may increase the accuracy in case of JW failing.

Nah, I won't make calls to the services via other add-ons on behalf of mine. Only for playback.
As I said, I already work on a solution that will filter the Search' results based on enabled services.
The Discover sections contain items not only offered by the enabled services - that's by design.

**Comment by bleak-lodge** [comment]
> Another idea for a future function:
> When scheduling a TV recording via a PVR addon, the whitelodge plugin could check whether the programme to be recorded is already available via one of the subscribed streaming services.

Sorry, but that's out of scope of this add-on. That's probably something a service plugin would do.

**Comment by bleak-lodge** [comment]
> 4\. Sometimes i get search results, but when i click it, it doesn't find any source on either netflix, disney, hbo ,prime . This is kind of confusing.

I've added new search functions under My Services and Search menus that'll filter the results based on user enabled services.

**Comment by yoshimo** [comment]
How well does this new search handle the following case?

Your main locale is german and most of your services are localised but as a foreign user you can also get services like hbo from another european country.

You of course want the proper results from both and not miss out the scandinavian hbo just because you set the whitelodge region to DE.

**Comment by bleak-lodge** [comment]
It works like the streams fetching mechanism - you only get results from content that is officially available on your enabled services in the country you've specified in the Country Code setting.
You want to get content available in other countries as well? That's not possible. You can use the the general search that does not filter results based on enabled services/country availability, but obviously the returned content is not guaranteed to have available streams.

**Comment by GG-76** [comment]
Hello, I installed your add-on because the current osmosis is no longer supported and gradually stops working. Unfortunately, I have two major problems to happy use.

1, I am unable to play some series, even if they are available on the selected VOD. An example is Black Mirror (Neflix) or  Futurama (Disney+). The only problem I can think of is that the individual series are numbered differently on VOD than on Trakt, but maybe the problem is something else.
2, are you thinking of adding SkyShowTime support for which there is a Kodi AddOn?

**Comment by bleak-lodge** [comment]
1. Have you set your country on add-on's settings/Providers tab? I can get Netflix link for Black Mirror, but can't tell if it works as I don't currently have an active Netflix sub. About Futurama, yeah there's a difference in how TMDb lists it (8 seasons) and how JustWatch does (14 seasons). For the 8 seasons that are listed in the add-on (as it relies on TMDb for seasons/episodes) I get streams for Hulu, not Disney+ (country code US).

2. Last time I checked this add-on works on Android only, so no - can't even test it.

**Comment by GG-76** [comment]
yes, I have set the correct country code for my country. Other series on Netflix are found. But not Black Mirror, nor can I find it through the search in the add-on. When I launch the official Netflix app, I find Black Mirror there.
As for Futurama, yes, I use Disney+, but in the end I found out that I can't play anything from Disney+, the result is always not found :( At the same time, Disney+ is in the list of supported services in my country code and I normally use it. So I don't know where it is problem...
Anyway, apart from that, is there any way to solve the problem with series where the numbering of the seasons is different compared to VOD? I know of several such cases, e.g. the Disenchantment series on Neflix.

**Comment by bleak-lodge** [comment]
Do you maybe have Kodi to hide stuff you've watched?
All this seems too vague to me, like what do you mean "not found",  not finding content at all (like movies/series) or not finding streams for content?
Specific country code and shows/movies would help me check if there is an actual problem - appart from tmdb/justwatch/actual provider content discrepancies which, no, can't be overtaken for now.

**Comment by GG-76** [comment]
I installed a clean install of kodi and the Disney+ and Whitelodge addon. I exported my watchlist from Trakt to the Kodi library and tried to run any of my watched series and they all end up with the same error:
1, a dialog will appear with a green selection of the Disney+ service, which is ok
2, an error message will appear: Disney+ error: "could not find stream for this content"
3. My country code is CZ
4, there are many series that you can try, the result is the same for all of them, e.g. Ahsoka, How I Met Your Father, Loki

**Comment by GG-76** [comment]
@bleak-lodge 
I tried to play a series from Disney+ directly without exporting  stream from WhiteLodge to the library and Kodi crashed completely with this error:

2024-02-24 17:05:13.438 T:8344    error <general>: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.whitelodge/?action=play&title=Aftermath&year=2021&imdb=tt12708542&tmdb=105971&season=1&episode=1&tvshowtitle=Star+Wars%3A+The+Bad+Batch&premiered=2021-05-04&meta=%7B%22title%22%3A+%22Aftermath%22%2C+%22label%22%3A+%22Aftermath%22%2C+%22season%22%3A+%221%22%2C+%22episode%22%3A+%221%22%2C+%22tvshowtitle%22%3A+%22Star+Wars%3A+The+Bad+Batch%22%2C+%22year%22%3A+%222021%22%2C+%22premiered%22%3A+%222021-05-04%22%2C+%22rating%22%3A+%227.941%22%2C+%22votes%22%3A+%2234%22%2C+%22director%22%3A+%22Steward+Lee%2C+Saul+Ruiz%2C+Nathaniel+Villanueva%22%2C+%22writer%22%3A+%22Jennifer+Corbett%2C+Dave+Filoni%22%2C+%22castwiththumb%22%3A+%5B%7B%22name%22%3A+%22Dee+Bradley+Baker%22%2C+%22role%22%3A+%22The+Bad+Batch+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F9oFnToDZWp0I484s7Ua1EzNQQ2m.jpg%22%7D%2C+%7B%22name%22%3A+%22Michelle+Ang%22%2C+%22role%22%3A+%22Omega+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F1RU0HHZQOXXO8ulThNxIyw0Ude1.jpg%22%7D%2C+%7B%22name%22%3A+%22Noshir+Dalal%22%2C+%22role%22%3A+%22Stormtrooper+%2F+Vice+Admiral+Rampart+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FkPZhR7ZxU5hnOimPZcJQMJ3qjoy.jpg%22%7D%2C+%7B%22name%22%3A+%22Liam+O%27Brien%22%2C+%22role%22%3A+%22Bolo+%2F+Tactical+Droid+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fko7Bba6fQEaAbVFAp1kQHTzuYCp.jpg%22%7D%2C+%7B%22name%22%3A+%22Sam+Riegel%22%2C+%22role%22%3A+%22Factory+Worker+%232+%2F+Ketch+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FfgQMEXtqK4xubDf0a2P8r4VJ8SF.jpg%22%7D%2C+%7B%22name%22%3A+%22Gwendoline+Yeo%22%2C+%22role%22%3A+%22Nala+Se+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FqcT51wtWxdS51q9FQcDC455HrqY.jpg%22%7D%2C+%7B%22name%22%3A+%22Rhea+Perlman%22%2C+%22role%22%3A+%22Cid+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F3kUrd75OesnsmicI15dWdswHy7T.jpg%22%7D%2C+%7B%22name%22%3A+%22Bob+Bergen%22%2C+%22role%22%3A+%22Lama+Su+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FsoDD1xiV3zVQWOJzVdJwfgIqzfe.jpg%22%7D%2C+%7B%22name%22%3A+%22Corey+Burton%22%2C+%22role%22%3A+%22Cad+Bane+%2F+Scrapper+%233+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FwcpFwdL7m01cZoBJiUSMQVakK2G.jpg%22%7D%2C+%7B%22name%22%3A+%22Ben+Diskin%22%2C+%22role%22%3A+%22AZI-3+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F4RWjcrPgfTG1yp19kFKRZYimv2B.jpg%22%7D%2C+%7B%22name%22%3A+%22Matthew+Wood%22%2C+%22role%22%3A+%22Battle+Droids+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fu1UjHLTeS5Dregg85A3GICf7Nkf.jpg%22%7D%2C+%7B%22name%22%3A+%22Stephen+Stanton%22%2C+%22role%22%3A+%22Admiral+Tarkin+%2F+Fighter+%231+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FuaoZXILATHT27HUnBh8Q345glc6.jpg%22%7D%2C+%7B%22name%22%3A+%22Daheli+Hall%22%2C+%22role%22%3A+%22ES-04+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FgUCLmhr5obfUamMFwgKJk2x3tTV.jpg%22%7D%2C+%7B%22name%22%3A+%22Tina+Huang%22%2C+%22role%22%3A+%22ES-02+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FyC7Pb1hEbe6wDbH1uprrKgpaNPp.jpg%22%7D%2C+%7B%22name%22%3A+%22Ness+Bautista%22%2C+%22role%22%3A+%22ES-03+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F9JrOV4lJlXCOaO6QG6wkq3eXbP6.jpg%22%7D%2C+%7B%22name%22%3A+%22Ming-Na+Wen%22%2C+%22role%22%3A+%22Fennec+Shand+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FeX7U4rXvXNsscog43qF02cPHUWI.jpg%22%7D%2C+%7B%22name%22%3A+%22Seth+Green%22%2C+%22role%22%3A+%22Todo+360+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fl4No5Eu6j0U80hCIkaSn17AOWrj.jpg%22%7D%2C+%7B%22name%22%3A+%22Shelby+Young%22%2C+%22role%22%3A+%22Captain+Bragg+%2F+Patron+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Foh4VeJpkORXoGJdIohNpXJyO3kt.jpg%22%7D%2C+%7B%22name%22%3A+%22Robin+Atkin+Downes%22%2C+%22role%22%3A+%22Cham+Syndulla+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fqb4xPIJGiKyiQ2MhSDQzFWxN0jS.jpg%22%7D%2C+%7B%22name%22%3A+%22Ferelith+Young%22%2C+%22role%22%3A+%22Eleni+Syndulla+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F420Nedauaegpa7xkBkqAOpOh4d5.jpg%22%7D%2C+%7B%22name%22%3A+%22Vanessa+Marshall%22%2C+%22role%22%3A+%22Hera+Syndulla+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FyZsWfaiOxaN5MZkWIKPbxz1KKS1.jpg%22%7D%2C+%7B%22name%22%3A+%22Dave+Filoni%22%2C+%22role%22%3A+%22C1-10P+a.k.a.+%27Chopper%27+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FdDTTrShpCnCacxB3LhpPffFIdB5.jpg%22%7D%2C+%7B%22name%22%3A+%22Archie+Panjabi%22%2C+%22role%22%3A+%22Depa+Billaba+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F1kSx9zUPwpTe35TbL7T2BWOp3bp.jpg%22%7D%2C+%7B%22name%22%3A+%22Freddie+Prinze+Jr.%22%2C+%22role%22%3A+%22Caleb+Dume+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FelcWqWi2aNZHDdfOZ21DISCCj08.jpg%22%7D%2C+%7B%22name%22%3A+%22Ian+McDiarmid%22%2C+%22role%22%3A+%22Darth+Sidious+%2F+Emperor+Palpatine+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FjeoA2UQdKoz5rp4xLv0O32gzJ38.jpg%22%7D%2C+%7B%22name%22%3A+%22Tom+Kane%22%2C+%22role%22%3A+%22Narrator+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fsq49aVP56G50Nr5kCL7Ove6Z9jE.jpg%22%7D%2C+%7B%22name%22%3A+%22Bonnie+Wild%22%2C+%22role%22%3A+%22PA+System+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FidPtIhDGE9IZ6w6ZTsKOIWZ0wKC.jpg%22%7D%2C+%7B%22name%22%3A+%22Andrew+Kishino%22%2C+%22role%22%3A+%22Saw+Gerrera+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2FdnQ6GYWWfogRnzapjzz7qgi5wPe.jpg%22%7D%2C+%7B%22name%22%3A+%22Nika+Futterman%22%2C+%22role%22%3A+%22Fauja+%2F+Shaeeah+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F6zfuSfLGtLXtFxXBUftMQbgfH6q.jpg%22%7D%2C+%7B%22name%22%3A+%22Kath+Soucie%22%2C+%22role%22%3A+%22Jek+%28voice%29%22%2C+%22thumbnail%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2F62hvT4USvGa9OAsMP9guIBtJqkf.jpg%22%7D%5D%2C+%22status%22%3A+%22Returning+Series%22%2C+%22plot%22%3A+%22The+clones+of+The+Bad+Batch+find+themselves+in+a+changing+galaxy+after+The+Clone+Wars.%22%2C+%22imdb%22%3A+%22tt12708542%22%2C+%22imdbnumber%22%3A+%22tt12708542%22%2C+%22tmdb%22%3A+%22105971%22%2C+%22unaired%22%3A+%22%22%2C+%22thumb%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2Fn4D7u3GEd7CwrTcdfovlddLhLjC.jpg%22%2C+%22poster%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2F71bdaIjnP5KXARZF8duF4X6ULu0.jpg%22%2C+%22fanart%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw1280%2Fxs2v0K7iKAuR4YtKOw7TEcDIX63.jpg%22%2C+%22banner%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw1280%2Fxs2v0K7iKAuR4YtKOw7TEcDIX63.jpg%22%2C+%22landscape%22%3A+%22https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw1280%2Fxs2v0K7iKAuR4YtKOw7TEcDIX63.jpg%22%2C+%22mediatype%22%3A+%22episode%22%2C+%22code%22%3A+%22105971%22%2C+%22trailer%22%3A+%22plugin%3A%2F%2Fplugin.video.whitelodge%2F%3Faction%3Dimdb_trailer%26name%3DStar%2BWars%253A%2BThe%2BBad%2BBatch%26tmdb%3D105971%26imdb%3Dtt12708542%26season%3D1%26episode%3D1%22%2C+%22duration%22%3A+%222700%22%7D&t=20240224160449619576]
2024-02-24 17:05:13.438 T:8344  warning <general>: Attempt to set unplayable index 0
2024-02-24 17:05:22.266 T:8344  critical <general>: Logic error due to two concurrent busydialogs, this is a known issue. The application will exit.

**Comment by bleak-lodge** [comment]
Disney+ was working when I first coded it, but a lot has been going on since then - mainly justwatch api changed to a new format which I adapted to, but can't test Disney+ as I currently don't have an active sub.
Couldn't reproduce the crash though, are you on Kodi 21 by any chance?

**Comment by GG-76** [comment]
@bleak-lodge yes I'm on Kodi 21. Is there any way I can help fix this? I would really appreciate it if your addon works with Disney+ because no other similar one works with Disney+

I have one other question, what is the best way to automatically synchronize my watchlist with Trakt to my Kodi library? Have i created my watchlist in Trakt and imported it manually into the library using your addon? when I add a new serial to the watchlist in the Trakt, is it automatically added to the kodi library? Or is there another better way, for example through automatic collection synchronization?

**Comment by bleak-lodge** [comment]
I haven't tested Kodi 21 yet.
I will check what's wrong with Disney+ when I renew my sub - can't really do anything else otherwise.
Trakt watchlist auto sync to library isn't supported, only collection.

**Comment by GG-76** [comment]
@bleak-lodge OK thank you for answer. I have one more question. in the description it is possible to play local files through this add-on, I haven't figured out how to do it. I understand that I can have something in the library that I don't play from VOD, but from local storage?

**Comment by bleak-lodge** [comment]
With local I mean from the library, ie if you scrape a movie from the add-on that also exists in your library (video file, not strm), it will be listed in the sources.

I checked Kodi 21 beta 3. Lots of changes that impact the add-on. I have fixed the crash on failed playback and some other quirks. I can also easily add sync trakt watchlist to library. Need to do some more work for Kodi 21, might also wait for the next beta/rc in case they revert some things before I release a fixed version, in the meantime consider k21 unsupported for the add-on.

**Comment by GG-76** [comment]
@bleak-lodge I understand that Kodi 21 is still a bit problematic, but it still has a lot of advantages. Anyway, I'll be really grateful for the Disney+ support fix and even more for the automatic watchlist update feature from the trakt. That would be really cool and if you need to test anything be sure to let me know.

Regarding the local files, I don't know if I understand it correctly, so I'll give an example. I have a series that I have in the watchlist of the trakt and I import it into the Kodi library, but it is not available anywhere on any VOD, but I have it available locally, but your add-on does not know this or can it be set somehow? Because now, when I try to play such a local serial from the Kodi library, of course the add-on tells me that no playback source was found. The second example can even be that one season of the series is available on VOD and one locally. Can this also be resolved?

anyway thank you very much for your add-on

**Comment by bleak-lodge** [comment]
I think you're having misunderstanding about how the add-on handles local (imported to library!) items and it's library integration function - and it's understandable.

The library source concerns playback from within the add-on's own menus. Ie if you search for sources for a movie/episode from within the add-on's own lists, and that said movie/episode exists in your Kodi library as a video file (not strm), the add-on will list the local  file for playback as a source along any other possible options for playback, ie a streaming service's add-on.

Now library integration function. The add-on will import item or items from within it's menus into your Kodi library if you choose to. It will exclude items that already exist in your local library as video files.
Ie if you choose to import a show from which you already have some seasons/episodes imported as video files, it will exclude those episodes.
It will also auto import new episodes as they air, granted the show is still airing.

Again, that is on supported Kodi versions (until k20), don't know how it performs on k21, as I said I saw many changes that impact the add-on's functionality. Also please check if Disney+ is working under k20.
I'm also eager for k21 for it's HDR support on Xbox, so once it reaches more stable versions I'll adapt, but as it is now I believe it will undergo more changes because many add-ons will be broken. I mean I already did some changes for it but I think it was a waste of time because things will change.

**Comment by GG-76** [comment]
@bleak-lodge thanks for explaining. For me, it's purely about the kodi library integration feature. I noticed that if I imported a serial from a trakt tv list that had already been imported into the library in whole or in part from local files, then whitelodge omitted them from the import (probably not 100%), but in principle it did. 
I would like to know how it will be preserved if, for example, I miss one season in the local, but it will be something that is not on VOD, when importing the trakt list, this missing season is exported as streams, is that right? What happens if I add the missing local files afterwards, will these be replaced instead of the streams when updating the library?

**Comment by bleak-lodge** [comment]
Well, that's something that you could easily try yourself and find out, but no an add-on has no business to replace library files lol

**Comment by GG-76** [comment]
@bleak-lodge I thought that if I add the missing local season and then have two sources in the library, i.e. streams imported before that, I will have to delete these streams manually, because the re-import from the watchlist at this moment already ignores this newly added local season, but the original streams will remain and this cannot be automated in any way, and the resulting duplication must be removed manually. Sorry, maybe stupid questions, I'm trying to understand how it all works, because it worked differently with osmosis. anyway, you're right, I can try it, I'm just trying to find, if possible, such a solution without manual intervention :)

**Comment by batarela** [comment]
![image](https://github.com/bleak-lodge/plugin.video.whitelodge/assets/3985100/1a75448b-4a67-4e24-9163-850cf1b7ebb9)

Google play store pushes KODI 21 officially, and plugin.video.whitelodge doesn't work anymore.

**Comment by ETE-Design** [comment]
@bleak-lodge What a nice addon :-) Would it be possible to add the content under "Videos / Tv-Shows" in the main menu on Kodi?
